src/views/game/ResultView.vue

```diff
@@ -18,11 +20,26 @@ const gameResultsDimensions = useElementBounding(gameResultsEl)
 const startingTransform = ref('')
 const displayTracks = ref(false)
 const displayButton = ref(false)
+const resultsAnimationPending = ref(true)
+const { width: screenWidth } = useWindowSize()
+const isMobileSize = computed(() => screenWidth.value < 1024)
 
 function handlePlayAgain() {
   router.push({ name: 'home' })
 }
 
+function handleResultsAnimationFinish() {
+  setTimeout(() => {
+    displayTracks.value = true
+  }, isMobileSize.value ? 200 : 1500)
```

Comment:
hokus pokus!
..wielkie rowery!
marcin ugotowal dziś magiczne numery

```
const animationDuration {
  FAST: 200
  MEDIUM: 800
  LONG: 1500
  MAX: 2000 // ew. inne nazwy i.e longest
} as const
```

Reply:
druga opcja to enum jak nie bedziesz po nim iterował, raczej pasuje dla tego use case

Reply:
`animationDuration` też nie powinno być MACRO_CASE?

Reply:
W sumie to raczej Pascal jeśli traktowac to jako enum

Reply:
Mam trochę wątpliwości co do tej sugestii, czas trwania w animacji CSS często deklaruję jako "magiczna liczba" (chyba że ty i tam tworzysz zmienną) czemu tutaj ma być inaczej

Reply:
A nawet jeśli bym wyciągnął to chyba bym próbował udawać Tailwinda z nazewnictwem, coś w stylu `d200`, `d500`, `d1500` itd.

Reply:
A czy musi mówić? `bg-primary`, `text-sm` tez za duzo nie mowią na pierwszy rzut oka, ale tworzysz sobie mentalna mapę w głowie i wiesz jaki to jest kolor/rozmiar czcionki.

Chodzi głowie tutaj o systematyzacje wartości w Twojej aplikacji. Stworzenie tokenów `200`, `500`, `1500` etc spoko pomysł, na ich podstawie możesz wtedy stworzyć aliasy, co jest dość często wykorzystywane w Design Systemach. np:
```
fast -> AnimationDuration.200
slow -> AnimationDuration.1500
```

, tak samo jak do koloru `primary` jest przypisywany pewien token z całej palety kolorów (w praktyce tak powinno być aby zachować spójność; taki system wprowadziłem w MP)
```
primary -> blue-500
foreground -> black
secondary -> lightblue-500
```

Im prostszy Design System tym jest łatwiejszy w obsludze i utrzymaniu. Jeśli korzystałbym sporo z animacji, to pewnie dobrą praktyką byłoby dodanie długości animacji do main.css (albo tailwind.css jesli to glowny plik css apki), które odwzorowywały by wartości z tego enuma. Jak trzeba będzie coś wydłużyć, skrócić, to zmieniają się jedynie 2 pliki.

Reply:
Możliwe, że są to scenariusze z większych aplikacji/zespołów, ale poniżej przedstawie sytuacje, które zdarzają się częściej niż można o tym pomyśleć:

Scenariusz 1
Element ma się pokazać szybko, spoko, importuje sobie enuma/consta, AnimationDuration.Fast, boom, z głowy.

Kurde, jednak chcialbym zeby wszystkie szybkie animacje trwały 150ms a nie 200ms, bo przeczytałem arytkuł na ten temat, że wtedy jest lepszy UX i wgl Apple tak robi, podmieniam wartosci w enumie i nie musze się marwić 10 wystąpieniami magicznych wartości.

Scenariusz 2
Do zespołu dochodzi nowa osoba, chce dodać animacje dla elementu. Ma być relatywnie szybka. Wchodzi w kod źródłowy GameResult, widzi magiczne wartości, dodaje kolejne magiczne wartości, bo bardziej jej pasuje, grafik wgl nic nie mówił i np. w codebase pojawia się 175ms, mimo tego ze wszedzie indziej jest 200ms. Marcin wyłapuje to w PR i mówi, ej lepiej by było gdyby to było 200ms, bo wszedzie tak mamy. Spoko, ale można było trochę czasu zaoszczędzić wrzucając to w pewny stały obiekt.

Reply:
Jeśli czujesz, że w ramach tego projektu to nie ma sensu, to jak najbardziej fair enough, może warto skipnąć, ja też w swoich apkach obieram pewne skróty, aczkolwiek czasami tego żałuję, także powyższe wiadomości traktuj jako swojego rodzj _cautionary tale_